src/main/java/io/stargate/sgv2/jsonapi/service/schema/collections/CollectionLexicalConfig.java

```diff
@@ -19,12 +20,38 @@ public record CollectionLexicalConfig(
   private static final JsonNode DEFAULT_NAMED_ANALYZER_NODE =
       JsonNodeFactory.instance.textNode(DEFAULT_NAMED_ANALYZER);
 
+  /**
+   * Constructs a lexical configuration with the specified enabled state and analyzer definition.
+   *
+   * <p>Validation behavior:
+   *
+   * <ul>
+   *   <li>If lexical search is enabled (enabled = true), the analyzer definition must not be null
+   *   <li>If lexical search is disabled (enabled = false), the analyzer definition must be null or
+   *       empty
+   * </ul>
+   *
+   * @param enabled Whether lexical search is enabled for this collection
+   * @param analyzerDefinition The JSON configuration for the analyzer, must not be null if lexical
+   *     search is enabled and must be null or empty if lexical search is disabled
+   * @throws NullPointerException if lexical search is enabled and analyzerDefinition is null
+   * @throws IllegalStateException if lexical search is disabled and analyzerDefinition is not null
+   */
   public CollectionLexicalConfig(boolean enabled, JsonNode analyzerDefinition) {
     this.enabled = enabled;
-    // Clear out any analyzer settings if not enabled (but don't fail)
     if (enabled) {
       this.analyzerDefinition = Objects.requireNonNull(analyzerDefinition);
     } else {
+      if (analyzerDefinition != null) {
+        if (analyzerDefinition.isTextual()) {
+          throw new IllegalArgumentException(
+              "Analyzer definition should not have string if lexical is disabled");
+        }
+        if (analyzerDefinition.isObject() && !analyzerDefinition.isEmpty()) {
```

Comment:
Not sure if any validation is needed here (since that is already done by caller), but if it is, should use similar logic -- if not enabled, must be `null` or empty Object.